tests/validation/tests/v3_api/test_hosted_eks_cluster.py

```diff
@@ -187,3 +223,177 @@ def get_logging_types():
+
+
+def edit_eks_cluster(cluster, eks_config_temp):
+    # edit eks_config_temp
+    # add new cloud cred
+    ec2_cloud_credential_new = get_aws_cloud_credential()
+    eks_config_temp["amazonCredentialSecret"] = ec2_cloud_credential_new.id
+    # add cluster level tags
+    eks_config_temp["tags"] = {"cluster-level-2": "tag2"}
+    # add node group
+    new_nodegroup = get_new_node()
+    eks_config_temp["nodeGroups"].append(new_nodegroup)
+    # remove all logging
+    eks_config_temp["loggingTypes"] = ["audit","api","authenticator"]
+    client = get_user_client()
+    client.update(cluster, name=cluster.name, eksConfig=eks_config_temp)
+    cluster = validate_cluster(client, cluster, intermediate_state="updating",
+                               check_intermediate_state=True,
+                               skipIngresscheck=True,
+                               timeout=DEFAULT_TIMEOUT_EKS)
+    return cluster
+
+
+def validate_nodegroup(nodegroup_list, cluster_name):
+    for nodegroup in nodegroup_list:
+        print("nodegroup:", nodegroup)
+        eks_nodegroup = AmazonWebServices().describe_eks_nodegroup(
+            cluster_name, nodegroup["nodegroupName"]
+        )
+        print("\nNode Group from EKS console: {}".format(eks_nodegroup))
+
+        # k8s version check
+        eks_cluster = AmazonWebServices().describe_eks_cluster(cluster_name)
+        assert eks_cluster["cluster"]["version"] == \
+               eks_nodegroup["nodegroup"]["version"], \
+            "Mismatch between K8s version of cluster and nodegroup"
+
+        # status of nodegroup
+        assert eks_nodegroup["nodegroup"]["status"] == "ACTIVE", \
+            "Nodegroups are not in active status"
+
+        # check scalingConfig
+        assert nodegroup["maxSize"] \
+               == eks_nodegroup["nodegroup"]["scalingConfig"]["maxSize"], \
+            "maxSize is incorrect on the nodes"
+        assert nodegroup["minSize"] \
+               == eks_nodegroup["nodegroup"]["scalingConfig"]["minSize"], \
+            "minSize is incorrect on the nodes"
+        assert nodegroup["minSize"] \
+               == eks_nodegroup["nodegroup"]["scalingConfig"]["minSize"], \
+            "minSize is incorrect on the nodes"
+
+        # check instance type
+        assert nodegroup["instanceType"] \
+               == eks_nodegroup["nodegroup"]["instanceTypes"][0], \
+            "instanceType is incorrect on the nodes"
+
+        # check disk size
+        assert nodegroup["diskSize"] \
+               == eks_nodegroup["nodegroup"]["diskSize"], \
+            "diskSize is incorrect on the nodes"
+        # check ec2SshKey
+        if "ec2SshKey" in nodegroup.keys():
+            assert nodegroup["ec2SshKey"] \
+                   == eks_nodegroup["nodegroup"]["remoteAccess"]["ec2SshKey"], \
+                "Ssh key is incorrect on the nodes"
+
+
+def wait_for_cluster_delete_eks(cluster_name, timeout=DEFAULT_TIMEOUT_EKS):
```

Comment:
Could we use an AWS (boto) waiter instead of our own custom?

there are a few waiters, I know we're using one for `nodegroups` there might be one too for cluster 

`waiter_ng = self._eks_client.get_waiter('nodegroup_deleted')`

Reply:
Oh you're deleting from Rancher, this might not be possible but I'm not completely sure.